tachyon/base/ref.h

```diff
@@ -38,6 +38,34 @@ class Ref {
       return *ref_ != *other.ref_;
     }
   }
+  bool operator<(const Ref& other) const {
+    if constexpr (SHALLOW) {
+      return ref_ < other.ref_;
+    } else {
+      return *ref_ < *other.ref_;
+    }
+  }
+  bool operator<=(const Ref& other) const {
+    if constexpr (SHALLOW) {
+      return ref_ <= other.ref_;
+    } else {
+      return *ref_ <= *other.ref_;
+    }
+  }
+  bool operator>(const Ref& other) const {
+    if constexpr (SHALLOW) {
+      return ref_ > other.ref_;
+    } else {
+      return *ref_ > *other.ref_;
+    }
+  }
+  bool operator>=(const Ref& other) const {
+    if constexpr (SHALLOW) {
+      return ref_ >= other.ref_;
+    } else {
+      return *ref_ >= *other.ref_;
+    }
+  }
```

Comment:
Shouldn't these changes be in 9641dd5?

Reply:
IMO, this change isn't part of `feat(base): differentiate DeepRef from ShallowRef`!

Reply:
@chokobole Then can you please remove the scope in 7b91f3b?